inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -1180,6 +1181,36 @@ template<>
 class ScaleFactorPerLayer<InferenceEngine::ConvolutionLayer*> : public ScaleFactorPerLayer<InferenceEngine::WeightableLayer*> {
 };
 
+template<>
+class ScaleFactorPerLayer<InferenceEngine::GemmLayer*> {
+public:
+    bool operator() (InferenceEngine::GemmLayer* gemmLayer, int weightsSize, int inputSize, ScaleFactorUpdateResult &result, const bool fakeQuantize) {
+        if ( !gemmLayer ) {
+            THROW_GNA_EXCEPTION << "Incorrect Gemm Layer pointer \n";
+        }
+        auto in0 = InferenceEngine::CNNNetPrevLayer(gemmLayer, 0);
+        auto in1 = InferenceEngine::CNNNetPrevLayer(gemmLayer, 1);
+
+        auto quantData = InferenceEngine::getInjectedData<QuantizedLayerParams>(*gemmLayer);
+        auto quantParams1 = InferenceEngine::getInjectedData<QuantizedLayerParams>(in1);
```

Comment:
Please, add some comments regarding Gemm inputs (which one must be const) and why we calculate weights scale factor from the second input which is dynamic for Gemm.